hugr-py/src/hugr/serialization/tys.py

```diff
@@ -87,12 +87,11 @@ def deserialize(self) -> tys.BoundedNatParam:
         return tys.BoundedNatParam(upper_bound=self.bound)
 
 
-class OpaqueParam(BaseTypeParam):
-    tp: Literal["Opaque"] = "Opaque"
-    ty: Opaque
+class StringParam(BaseTypeParam):
+    tp: Literal["String"] = "String"
 
     def deserialize(self) -> tys.OpaqueParam:
-        return tys.OpaqueParam(ty=self.ty.deserialize())
+        return tys.OpaqueParam()
```

Comment:
It does seem odd not to be returning something called `tys.StringParam` here. I won't 100% insist (if there are serious migration issues) but I will strongly encourage